### DISCUSSION TITLE
Understanding pidbox queue names

### DISCUSSION THREAD

**Discussion by rgov** [celery/kombu#1610] [Q&A]
I'm trying to debug a piece of third-party software that uses Celery (and therefore, indirectly, Kombu) for a remote control mailbox. I apologize for asking RTFM-able questions, but I spent a bit of time and could not unravel these.

It looks like somewhere in the stack a queue is created with a name like `celery@hostname.reply.celery.pidbox`. Note that this includes my computer's hostname, but some references I've found seem to use UUIDs instead of the hostname.

1. What is the `celery@` part called / referring to? Actually what are all the parts of this referring to? It is kind of un-Googleable.
2. Is there a reason to override the `celery@` (e.g., to associate it with the application user associated with the queue)? If so, how?
3. How do I override the use of the hostname?
4. In what cases is a UUID used instead of the hostname?
5. What is responsible for generating these names, Kombu or Celery? Does it vary based on the selected backend (e.g., RabbitMQ vs Redis)?

(Edit: I didn't see that discussions are enabled for this repo, feel free to transfer it there, or to the celery/celery repo if that's more appropriate.)

[a comment by an automated account is not shown]

**Comment by michaelkl** [comment]
I came across similar question.
Found [this article](https://www.ancisoft.com/blog/unknown-queue-names-show-on-rabbitmq-mgmt-when-using-celery/) regarding the topic briefy describing what is what. Queue naming does not quite match that in the question but it might be due to version differences over the years.
